src/averageAges.js

```diff
@@ -52,7 +57,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person =>
+    (onlyWithSon === undefined && person.mother !== null)
+    || (onlyWithSon === true
+        && person.mother !== null
+        && person.sex === 'm')
+  );
+
+  const mothersAndChildren = children.map(child => {
+    return {
+      mom: people.find(mom => mom.name === child.mother),
+      child: child,
+    };
+  }).filter(momAndChild => momAndChild.mom !== undefined);
+
+  return mothersAndChildren.map(momAndChild =>
```

Comment:
Не переносимо рядок після стрілки. Переноситься або все разом, або пишемо все в одну строку
```suggestion
  return mothersAndChildren.map(
    momAndChild => momAndChild.child.born - momAndChild.mom.born
  )
```